### PR TITLE
fixed uncaught exception when tag or title

### PR DESCRIPTION
**Pull request by callmewa**
fixed uncaught exception when tag or title (both optional according to api) is not provided for upload.

updated test accordingly

### CODE REVIEW

**Comment by callmewa** [comment]
ran into this problem during development.. hopefully others can benefit from this change

**Comment by Pomax** [comment]
While I can understand uploading an image without tags, uploading an image without a title makes no sense (and I'm pretty sure the FlickrAPI throws an error if you try to), so it's pretty important that a missing title causes an error *somewhere*, even if this is not the right place for it. There should always be an options.title, so if this patch gets to land, it needs to at the very least also contain a fix somewhere so that the default title for an upload, if there is no explicit title, is the photo's filename.

**Comment by callmewa** [comment]
Here is flickr's rest api options for uploading a photo.  We should not place additional restrictions on the flexibility the API offers us.  https://www.flickr.com/services/api/upload.api.html
Depending on what you are trying to do with your application it makes perfect sense not to have a title for the photo.  
title (optional)
The title of the photo.
description (optional)
A description of the photo. May contain some limited HTML.
tags (optional)
A space-seperated list of tags to apply to the photo.

File: test.js [hunk 1]
```diff
@@ -104,9 +104,7 @@ if(testAuthenticated) Flickr.authenticate(FlickrOptions, function(error, flickr)
         var uploadOptions = {
           photos: [
             {
-              title: "test",
-              photo: __dirname + "/test.jpg",
-              tags: ['happy fox', 'test 1']
+              photo: __dirname + "/test.jpg"
```
Comment (Pomax) [inline test.js]: please add a case here, instead of rewriting one.

**Comment by Pomax** [comment]
fair enough - please add your testcase as new case though, don't rewrite the ones that were already there.

**Comment by callmewa** [comment]
Just wondering if the last commit was acceptable.  Thanks

**Comment by Pomax** [comment]
yeah, github just decided to not notify me that a new commit had been pushed. Looks good to me.

**Pull request merged**